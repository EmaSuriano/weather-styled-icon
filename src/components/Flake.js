import React from 'react';
import styled from 'styled-components';
import Spin from './animations/Spin';
import { ThemePropType, defaultTheme } from '../constants';

const Flake = styled.div.attrs({
  'data-e2e': 'flake',
  'data-animation': true,
})`
  /* position: absolute; */

    color: ${props => props.theme.snowColor};
    opacity: 0.4;
    top: 50%;
  left: 50%;
  width: 1em;
  height: 1em;
    animation: ${Spin} 8s linear infinite ${props =>
  props.reverse ? 'reverse' : ''};

   /* &:before,
  &:after {
    content: '\\2744';
    position: absolute;

    opacity: 0.4;
  } */
  /*

  &:after {
    margin: -0.25em 0 0 -1em;
    font-size: 1.5em;
    opacity: 0.6;
    animation: ${Spin} 14s linear infinite;
  } */

  /* &:nth-child(2):before {
    margin: -0.75em 0 0 0.15em;
    font-size: 1.25em;
    opacity: 0.4;
    animation: ${Spin} 10s linear infinite;
  }

  &:nth-child(2):after {
    margin: 0em 0 0 0.125em;
    font-size: 2em;
    opacity: 0.6;
    animation: ${Spin} 16s linear infinite reverse;
  } */
`;

const A = styled.div`
  display: flex;
`;

const Flakes = () => (
  <A>
    <Flake>&#10052;</Flake>
    <Flake reverse>&#10052;</Flake>
  </A>
);

Flakes.displayName = 'Flakes';

export default Flakes;
