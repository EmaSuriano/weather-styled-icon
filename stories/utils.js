import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { WeatherThemeProvider } from '../src';
import { ThemePropType } from '../src/constants';

export const CenterContainer = styled.div`
  ${props =>
    props.theme.backgroundColor &&
    `background-color: ${props.theme.backgroundColor}`};
  text-align: center;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const BackgroundContainer = styled(CenterContainer)`
  background-color: ${props => props.color};
`;

export const ThemeContainer = ({ theme, children }) => (
  <WeatherThemeProvider theme={theme}>
    <BackgroundContainer>{children}</BackgroundContainer>
  </WeatherThemeProvider>
);

ThemeContainer.propTypes = {
  theme: ThemePropType.isRequired,
  children: PropTypes.node,
};

ThemeContainer.defaultProps = {
  children: PropTypes.node.isRequired,
};
