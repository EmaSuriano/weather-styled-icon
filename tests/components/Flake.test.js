import React from 'react';
import Flake from '../../src/components/Flake';
import { renderWithTheme } from '../utils';

describe('<Flake />', () => {
  it('should should render div with theme', () => {
    const wrapper = renderWithTheme(<Flake />);

    expect(wrapper).toMatchSnapshot();
  });

  xit('Spin should return keyframes with theme', () => {
    // TOOD: Find a proper way to test Animations!
  });
});
