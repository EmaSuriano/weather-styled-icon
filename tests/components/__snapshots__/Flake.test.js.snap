// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flake /> should should render div with theme 1`] = `
.c0:before,
.c0:after {
  content: '\\2744';
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -1.5em 0 0 -1.0125em;
  color: white;
  opacity: 0.4;
  -webkit-animation: cZxgpV 8s linear infinite reverse;
  animation: cZxgpV 8s linear infinite reverse;
}

.c0:after {
  margin: -0.25em 0 0 -1em;
  font-size: 1.5em;
  opacity: 0.6;
  -webkit-animation: cZxgpV 14s linear infinite;
  animation: cZxgpV 14s linear infinite;
}

.c0:nth-child(2):before {
  margin: -0.75em 0 0 0.15em;
  font-size: 1.25em;
  opacity: 0.4;
  -webkit-animation: cZxgpV 10s linear infinite;
  animation: cZxgpV 10s linear infinite;
}

.c0:nth-child(2):after {
  margin: 0em 0 0 0.125em;
  font-size: 2em;
  opacity: 0.6;
  -webkit-animation: cZxgpV 16s linear infinite reverse;
  animation: cZxgpV 16s linear infinite reverse;
}

<div
  className="c0"
/>
`;
